### PR TITLE
fix(typescript-estree): fix more cases with double slash in JSX text

### LINKED ISSUE #606: Parser mishandling comments in jsx including //.
```ts
const parser = require('@typescript-eslint/typescript-estree');
const code = `
  <div>
    <div>hello</div>
    // This is not a comment.
  </div>
`;
const ast = parser.parse(code, {
    jsx: true,
    comment: true,
});
return ast.comments;
```

**What did you expect to happen?**
`[]` is returned.

**What actually happened?**
`[ { type: 'Line',
    value: ' This is not a comment.',
    range: [ 34, 59 ],
    loc: { start: [Object], end: [Object] } } ]` is returned.
**Versions**

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/typescript-estree` | `1.10.2` |
| `TypeScript`                           | `3.5.1` |
| `node`                                 | `11.9.0` |
| `npm`                                  | `6.9.0` |

### PR DESCRIPTION
**Pull request by duailibe**
This improves upon the commit 72552a33d6266f55bd25f0e8abd91732a183e33a, catching more cases that needs to call `rescanJsxToken()`: after a closing element, after a self-closing element and after a JSX expression, when inside a JSX element.

Caught this with Prettier: https://github.com/prettier/prettier/issues/6084

Fixes #606

### CODE REVIEW

**Comment by duailibe** [comment]
PS: looks like the following files need to be formatted
```
CHANGELOG.md
packages/eslint-plugin-tslint/CHANGELOG.md
packages/eslint-plugin/CHANGELOG.md
packages/experimental-utils/CHANGELOG.md
packages/parser/CHANGELOG.md
packages/typescript-estree/CHANGELOG.md
```

File: packages/typescript-estree/tests/lib/comments.ts [hunk 1]
```diff
@@ -14,6 +14,10 @@ const testFiles = glob.sync(`${FIXTURES_DIR}/**/*.src.*`);
 
 describe('Comments', () => {
   testFiles.forEach(filename => {
+    // let it = test.skip;
```
Comment (bradzacher) [inline packages/typescript-estree/tests/lib/comments.ts]: need to delete this

Comment (duailibe) [reply packages/typescript-estree/tests/lib/comments.ts]: Oops sorry!

**Comment by duailibe** [comment]
Removed the `isJsxOpeningTag` declaration 🤷‍♂

**Comment by duailibe** [comment]
I can't make CI green.. looks like there's some cache, it's no "seeing" the latest branch

**Comment by duailibe** [comment]
Nope, it was just me being dumb 😅

**Pull request merged**